File: app/lib/domain/models/day_task.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'exercise.dart';
import 'meal.dart';

part 'day_task.freezed.dart';

@freezed
class Workout with _$Workout {
  const factory Workout({
    required String title,
    required int estimatedMinutes,
    required List<Exercise> exercises,
  }) = _Workout;
}

@freezed
class DayTask with _$DayTask {
  const factory DayTask({
    int? dbId,                          // 仅从 DB 读出时填，用于 markWorkoutDone/markMealDone
    required int planId,
    required int dayIndex,
    required DateTime date,
    Workout? workout,
    @Default([]) List<Meal> meals,
    @Default(false) bool completedWorkout,
    @Default({}) Map<String, bool> completedMeals,
    @Default(0) int xpAwarded,
    DateTime? completedAt,
  }) = _DayTask;
}

File: app/android/app/src/main/java/com/gymcoach/gym_coach/domain/model/DayTask.kt
package com.gymcoach.gym_coach.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDate

/**
 * Mirrors `lib/domain/models/day_task.dart`. Stored split across several DB
 * fields (`workout_json`, `meals_json`, `completed_workout`, etc.) — the
 * composite shape is only held in memory. `dbId` is null when a task is
 * fabricated in memory (not yet persisted).
 */
data class DayTask(
    val dbId: Int? = null,
    val planId: Int,
    val dayIndex: Int,
    val date: LocalDate,
    val workout: Workout? = null,
    val meals: List<Meal> = emptyList(),
    val completedWorkout: Boolean = false,
    val completedMeals: Map<String, Boolean> = emptyMap(),
    @SerialName("xp_awarded") val xpAwarded: Int = 0,
    val completedAt: Long? = null
)
